### PR TITLE
feature/560 - Add polling capability for generated .fec file task

### PR DESCRIPTION
**Pull request by sasha-dresden**
Issue [#560](https://github.com/fecgov/fecfile-web-app/issues/560) 

APP [PR1883](https://github.com/fecgov/fecfile-web-app/pull/1883)

### CODE REVIEW

**Review by Elaine-Krauss-TCG — APPROVED** [review]

File: django-backend/fecfiler/web_services/views.py [hunk 1]
```diff
@@ -37,26 +41,38 @@ def create_dot_fec(self, request):
...
+    def get_dot_fec(self, request, id):
         """Download the most recent .FEC created for a report
         Currently only useful for testing purposes
         """
-        committee_uuid = request.session["committee_uuid"]
-        dot_fec_record = DotFEC.objects.filter(
-            report__id=report_id, report__committee_account_id=committee_uuid
-        ).order_by("-file_name")
+
+        dot_fec_record = DotFEC.objects.filter(id=id)
```
Comment (toddlees) [inline django-backend/fecfiler/web_services/views.py:73]: We still need to filter by `report__committee_account_id=committee_uuid` to limit access to your committee

**Review by toddlees — CHANGES_REQUESTED** [review]

File: django-backend/fecfiler/web_services/views.py (hunk 1 above)
Comment (sasha-dresden) [reply django-backend/fecfiler/web_services/views.py]: Ah that makes sense. Got that added back in.

File: django-backend/fecfiler/web_services/views.py [hunk 2]
```diff
@@ -37,26 +41,40 @@ def create_dot_fec(self, request):
...
+    @action(
+        detail=False,
+        methods=["get"],
+        url_path="dot-fec/check/(?P<task_id>[a-z0-9-]+)",
+    )
+    def check_dot_fec(self, request, task_id):
+        res = AsyncResult(task_id)
+        if res.ready():
+            return Response({"done": True, "id": res.get()})
+        return Response({"done": False})
```
Comment (mjtravers) [inline django-backend/fecfiler/web_services/views.py:60]: Great way to check if ready.

**Review by mjtravers — APPROVED** [review]

File: django-backend/fecfiler/web_services/views.py [hunk 3]
```diff
@@ -37,26 +41,40 @@ def create_dot_fec(self, request):
...
 
     @action(
         detail=False,
         methods=["get"],
-        url_path="dot-fec/(?P<report_id>[a-z0-9-]+)",
+        url_path="dot-fec/(?P<id>[a-z0-9-]+)",
         renderer_classes=(DotFECRenderer,),
     )
-    def get_dot_fec(self, request, report_id):
```
Comment (toddlees) [inline django-backend/fecfiler/web_services/views.py:50]: I like this change to cut the report out or the equation

**Review by toddlees — APPROVED** [review]

**Pull request merged**